platforms/cpuinfo.go

```diff
@@ -43,6 +45,32 @@ func cpuVariant() string {
 	return cpuVariantValue
 }
 
+// get CPU Architecture from uname -m
+func getCPUArchitecture() (string, error) {
+	if !isLinuxOS(runtime.GOOS) {
+		return "", fmt.Errorf("getCPUArchitecture for OS %s: %w", runtime.GOOS, errdefs.ErrNotImplemented)
+	}
+	var name unix.Utsname
+	err := unix.Uname(&name)
+	if err != nil {
+		return "", err
+	}
+	machine := string(name.Machine[:bytes.IndexByte(name.Machine[:], 0)])
+	// https://github.com/qemu/qemu/blob/master/linux-user/uname.c
```

Comment:
Does it depend QEMU?

Reply:
Not exactly, but the method does assume the machine field returned from syscall does match armvXl or armvXb pattern.

Reply:
Then you should mention that instead of having the link to QEMU.